Library/Homebrew/test/cask/cli/upgrade_spec.rb

```diff
@@ -0,0 +1,189 @@
+      expect(Hbc.appdir.join("Caffeine.app")).to be_a_directory
+      expect(Hbc::CaskLoader.load("local-caffeine").versions).to include("1.2.3")
+
+      expect(Hbc::CaskLoader.load("auto-updates")).to be_installed
+      expect(Hbc.appdir.join("MyFancyApp.app")).to be_a_directory
+      expect(Hbc::CaskLoader.load("auto-updates").versions).to include("2.61")
+    end
+  end
+
+  describe "with --greedy it checks additional Casks" do
+    include_context "Proper Casks"
+
+    it 'includes the Casks with "auto_updates true" or "version latest" with --greedy' do
+      expect(Hbc::CaskLoader.load("auto-updates")).to be_installed
+      expect(Hbc.appdir.join("MyFancyApp.app")).to be_a_directory
+      expect(Hbc::CaskLoader.load("auto-updates").versions).to include("2.57")
+
+      expect(Hbc::CaskLoader.load("local-caffeine")).to be_installed
+      expect(Hbc.appdir.join("Caffeine.app")).to be_a_directory
+      expect(Hbc::CaskLoader.load("local-caffeine").versions).to include("1.2.2")
+
+      expect(Hbc::CaskLoader.load("local-transmission").versions).to include("2.60")
+
+      described_class.run("--greedy")
+
+      expect(Hbc::CaskLoader.load("auto-updates")).to be_installed
+      expect(Hbc.appdir.join("MyFancyApp.app")).to be_a_directory
+      expect(Hbc::CaskLoader.load("auto-updates").versions).to include("2.61")
+
+      expect(Hbc::CaskLoader.load("local-caffeine")).to be_installed
+      expect(Hbc.appdir.join("Caffeine.app")).to be_a_directory
+      expect(Hbc::CaskLoader.load("local-caffeine").versions).to include("1.2.3")
+
+      expect(Hbc::CaskLoader.load("local-transmission")).to be_installed
+      expect(Hbc.appdir.join("Transmission.app")).to be_a_directory
+      expect(Hbc::CaskLoader.load("local-transmission").versions).to include("2.61")
+    end
+
+    it 'does not include the Casks with "auto_updates true" when the version did not change' do
+      expect(Hbc::CaskLoader.load("auto-updates")).to be_installed
+      expect(Hbc.appdir.join("MyFancyApp.app")).to be_a_directory
+      expect(Hbc::CaskLoader.load("auto-updates").versions).to include("2.57")
+
+      described_class.run("auto-updates", "--greedy")
+
+      expect(Hbc::CaskLoader.load("auto-updates")).to be_installed
+      expect(Hbc.appdir.join("MyFancyApp.app")).to be_a_directory
+      expect(Hbc::CaskLoader.load("auto-updates").versions).to include("2.61")
+
+      described_class.run("auto-updates", "--greedy")
+
+      expect(Hbc::CaskLoader.load("auto-updates")).to be_installed
+      expect(Hbc.appdir.join("MyFancyApp.app")).to be_a_directory
+      expect(Hbc::CaskLoader.load("auto-updates").versions).to include("2.61")
+    end
+  end
+
+  describe "handles borked upgrades" do
+    include_context "Casks that will fail upon upgrade"
+
+    output_reverted = Regexp.new <<~EOS
+      Warning: Reverting upgrade for Cask .*
+    EOS
+
+    it "by restoring the old Cask if the upgrade's install failed" do
```

Comment:
“restores” instead of “by restoring” (Basically all of these should read like full sentences.)

Reply:
Maybe also just “upgrade“ instead of “upgrade's install”.